### ISSUE TITLE
Fails with clang: non-constant-expression cannot be narrowed from type 'unsigned long' to 'float' in initializer list

### ISSUE THREAD

**Issue by yurivict** [LLNL/TopoMS#4]
```
In file included from /usr/ports/science/topoms/work/TopoMS-1.1/topoms/src/TopoMS-vtk.cpp:78:
/usr/ports/science/topoms/work/TopoMS-1.1/topoms/include/TopoMS.h:295:25: error: non-constant-expression cannot be narrowed from type 'unsigned long' to 'float' in initializer list [-Wc++11-narrowing]
        float cut[3] = {m_metadata.m_grid_dims[0], m_metadata.m_grid_dims[1], m_metadata.m_grid_dims[2]};
                        ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/science/topoms/work/TopoMS-1.1/topoms/include/TopoMS.h:295:25: note: insert an explicit cast to silence this issue
        float cut[3] = {m_metadata.m_grid_dims[0], m_metadata.m_grid_dims[1], m_metadata.m_grid_dims[2]};
                        ^~~~~~~~~~~~~~~~~~~~~~~~~
                        static_cast<float>(      )
/usr/ports/science/topoms/work/TopoMS-1.1/topoms/include/TopoMS.h:295:52: error: non-constant-expression cannot be narrowed from type 'unsigned long' to 'float' in initializer list [-Wc++11-narrowing]
        float cut[3] = {m_metadata.m_grid_dims[0], m_metadata.m_grid_dims[1], m_metadata.m_grid_dims[2]};
                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/science/topoms/work/TopoMS-1.1/topoms/include/TopoMS.h:295:52: note: insert an explicit cast to silence this issue
        float cut[3] = {m_metadata.m_grid_dims[0], m_metadata.m_grid_dims[1], m_metadata.m_grid_dims[2]};
                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~
                                                   static_cast<float>(      )
/usr/ports/science/topoms/work/TopoMS-1.1/topoms/include/TopoMS.h:295:79: error: non-constant-expression cannot be narrowed from type 'unsigned long' to 'float' in initializer list [-Wc++11-narrowing]
        float cut[3] = {m_metadata.m_grid_dims[0], m_metadata.m_grid_dims[1], m_metadata.m_grid_dims[2]};
                                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/ports/science/topoms/work/TopoMS-1.1/topoms/include/TopoMS.h:295:79: note: insert an explicit cast to silence this issue
        float cut[3] = {m_metadata.m_grid_dims[0], m_metadata.m_grid_dims[1], m_metadata.m_grid_dims[2]};
                                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~
                                                                              static_cast<float>(      )
3 errors generated.
[6/10] /usr/local/bin/clang++70   -I/usr/ports/science/topoms/work/TopoMS-1.1/msc/include -I/usr/ports/science/topoms/work/TopoMS-1.1/topoms/include -std=c++0x -fpermissive -w         -fopenmp=libomp -O2 -pipe -fno-omit-frame-pointer -fstack-protector -fno-strict-aliasing -fno-omit-frame-pointer -MD -MT topoms-cli/CMakeFiles/topoms-cli.dir/src/main.cpp.o -MF topoms-cli/CMakeFiles/topoms-cli.dir/src/main.cpp.o.d -o topoms-cli/CMakeFiles/topoms-cli.dir/src/main.cpp.o -c /usr/ports/science/topoms/work/TopoMS-1.1/topoms-cli/src/main.cpp
```

**Comment by yurivict** [comment]
```
bin/topoms-cli
lib/libtopoms.a
```

You probably intended to install more files.

**Comment by bhatiaharsh** [comment]
I can fix the typecasting error, but I am not sure what you mean by installing more files. The script installs the library (*.a), topoms-cli, and topoms-ui (if you choose). Can you clarify what you mean?

**Comment by yurivict** [comment]
Library can't be used without headers. Installing just a library, and no headers, is meaningless.

**Comment by bhatiaharsh** [comment]
That is not the intended purpose of this install. The library is being used only in the compiled executables.

typecasting fix: e53ebe2e1033b9126da150c262e8a9ee999e0f15

**Comment by yurivict** [comment]
Static libraries can't normally be used by executables.
